# Investment Case Group Project

In [1]:
# importing numpy and pandas libraries
import numpy as np
import pandas as pd

## Checkpoint 1 - Data Cleaning 1

###  1. Load the companies and rounds data (provided on the previous page) into two data frames and name them companies and rounds2 respectively.

In [2]:
# loading the data from the files provided to dataframes
companies = pd.read_csv("datafiles\companies.txt", sep='\t', header=0, encoding = "ISO-8859-1")
rounds2 = pd.read_csv("datafiles/rounds2.csv", encoding = "ISO-8859-1")

In [3]:
# resolving the encoding issue
companies['permalink'] = companies[['permalink']].applymap(lambda link:link.encode('utf-8').decode('ascii', 'ignore'))
rounds2['company_permalink'] = rounds2[['company_permalink']].applymap(lambda link:link.encode('utf-8').decode('ascii', 'ignore'))

In [4]:
companies.head(100)

permalink  \
0                        /Organization/-Fame   
1                     /Organization/-Qounter   
2        /Organization/-The-One-Of-Them-Inc-   
3                      /Organization/0-6-Com   
4             /Organization/004-Technologies   
5           /Organization/01Games-Technology   
6        /Organization/0Ndine-Biomedical-Inc   
7                       /Organization/0Xdata   
8                            /Organization/1   
9                  /Organization/1-2-3-Listo   
10                     /Organization/1-4-All   
11            /Organization/1-618-Technology   
12               /Organization/1-800-Dentist   
13               /Organization/1-800-Doctors   
14  /Organization/1-800-Publicrelations-Inc-   
15                /Organization/1-Mainstream   
16                     /Organization/1-Of-99   
17                 /Organization/10-20-Media   
18             /Organization/10-Minutes-With   
19                  /Organization/1000-Corks   
20                /Organization/1000-Markets   
21                     /Organization/1000Chi   
22                   /Organization/1000Lookz   
23                /Organization/1000Memories   
24             /Organization/1000Museums-Com   
25                  /Organization/1001-Menus   
26          /Organization/1001Pharmacies-Com   
27                     /Organization/1006-Tv   
28                    /Organization/100Du-Tv   
29                    /Organization/100E-Com   
..                                       ...   
70                        /Organization/1417   
71                       /Organization/15Fen   
72                      /Organization/15Five   
73                /Organization/15Minutesnow   
74                  /Organization/15Shopstop   
75           /Organization/16-Mile-Solutions   
76                      /Organization/169-St   
77                  /Organization/16Lab-Inc-   
78                      /Organization/16Wifi   
79                    /Organization/17-Media   
80                 /Organization/170-Systems   
81                      /Organization/17Hats   
82                      /Organization/17U-Cn   
83                     /Organization/17Zuoye   
84                 /Organization/1800Diapers   
85                /Organization/180Solutions   
86                        /Organization/1871   
87                       /Organization/19Pay   
88                        /Organization/1Bib   
89                        /Organization/1Bog   
90                  /Organization/1C-Company   
91                   /Organization/1Calendar   
92                       /Organization/1Cast   
93                      /Organization/1Click   
94             /Organization/1Cloudstar-Asia   
95                /Organization/1Commmedical   
96                     /Organization/1D4-Pty   
97                   /Organization/1Day1Song   
98                   /Organization/1Daylater   
99                /Organization/1Daymakeover   

                                   name                        homepage_url  \
0                                 #fame                  http://livfame.com   
1                              :Qounter              http://www.qounter.com   
2                (THE) ONE of THEM,Inc.                 http://oneofthem.jp   
3                               0-6.com                  http://www.0-6.com   
4                      004 Technologies   http://004gmbh.de/en/004-interact   
5                    01Games Technology              http://www.01games.hk/   
6                Ondine Biomedical Inc.                http://ondinebio.com   
7                                H2O.ai                      http://h2o.ai/   
8                              One Inc.                  http://whatis1.com   
9                           1,2,3 Listo             http://www.123listo.com   
10                              1-4 All                                 NaN   
11                     1.618 Technology  http://www.Homeandcondogallery.com   
12                        1-800

In [5]:
rounds2.head(100)

company_permalink  \
0                        /organization/-fame   
1                     /ORGANIZATION/-QOUNTER   
2                     /organization/-qounter   
3        /ORGANIZATION/-THE-ONE-OF-THEM-INC-   
4                      /organization/0-6-com   
5             /ORGANIZATION/004-TECHNOLOGIES   
6           /organization/01games-technology   
7        /ORGANIZATION/0NDINE-BIOMEDICAL-INC   
8        /organization/0ndine-biomedical-inc   
9                       /ORGANIZATION/0XDATA   
10                      /organization/0xdata   
11                      /ORGANIZATION/0XDATA   
12                      /organization/0xdata   
13                           /ORGANIZATION/1   
14                           /organization/1   
15                           /ORGANIZATION/1   
16                 /organization/1-2-3-listo   
17                     /ORGANIZATION/1-4-ALL   
18            /organization/1-618-technology   
19               /ORGANIZATION/1-800-DENTIST   
20               /organization/1-800-doctors   
21  /ORGANIZATION/1-800-PUBLICRELATIONS-INC-   
22                /organization/1-mainstream   
23                     /ORGANIZATION/1-OF-99   
24                 /organization/10-20-media   
25                 /ORGANIZATION/10-20-MEDIA   
26                 /organization/10-20-media   
27                 /ORGANIZATION/10-20-MEDIA   
28             /organization/10-minutes-with   
29             /ORGANIZATION/10-MINUTES-WITH   
..                                       ...   
70                       /organization/10six   
71                  /ORGANIZATION/10TRACKS-2   
72                  /organization/10tracks-2   
73                /ORGANIZATION/10X-GENOMICS   
74            /organization/10x-technologies   
75            /ORGANIZATION/10X-TECHNOLOGIES   
76                  /organization/10x10-room   
77                     /ORGANIZATION/10NORTH   
78                   /organization/11-health   
79           /ORGANIZATION/115-NETWORK-DISKS   
80                       /organization/117go   
81               /ORGANIZATION/11I-SOLUTIONS   
82                     /organization/12-labs   
83            /ORGANIZATION/12-STAR-SURVIVAL   
84                  /organization/120-sports   
85                 /ORGANIZATION/121-RENTALS   
86                     /organization/121cast   
87                     /ORGANIZATION/121CAST   
88                     /organization/121cast   
89                    /ORGANIZATION/121NEXUS   
90                    /organization/121nexus   
91                    /ORGANIZATION/121NEXUS   
92                    /organization/121nexus   
93                    /ORGANIZATION/121NEXUS   
94                    /organization/121nexus   
95                   /ORGANIZATION/1234ENTER   
96                   /organization/1234enter   
97              /ORGANIZATION/123CONTACTFORM   
98              /organization/123contactform   
99                 /ORGANIZATION/123FENG-COM   

                            funding_round_permalink   funding_round_type  \
0   /funding-round/9a01d05418af9f794eebff7ace91f638              venture   
1   /funding-round/22dacff496eb7acb2b901dec1dfe5633              venture   
2   /funding-round/b44fbb94153f6cdef13083530bb48030                 seed   
3   /funding-round/650b8f704416801069bb178a1418776b              venture   
4   /funding-round/5727accaeaa57461bd22a9bdd945382d              venture   
5   /funding-round/1278dd4e6a37fa4b7d7e06c21b3c1830              venture   
6   /funding-round/7d53696f2b4f607a2f2a8cbb83d01839          undisclosed   
7   /funding-round/2b9d3ac293d5cdccbecff5c8cb0f327d                 seed   
8   /funding-round/954b9499724b946ad8c396a57a5f3b72              venture   
9   /funding-round/383a9bd2c04f7038bb543ccef5ba3eae                 seed   
10  /funding-round/3bb2ee4a2d89251a10aaa735b1180e44              venture   
11  /funding-round/ae2a174c06517c2394aed45006322a7e              venture   
12  /funding-round/e1cfcbe1bdf4c70277c5f29a3482f24e           

### 2. Table 1.1:

#### 2.1 How many unique companies are present in rounds2?


In [6]:
# counting unique companies in rounds2 df
rounds2['company_permalink'].str.lower().nunique()

66368

#### 2.2 How many unique companies are present in companies?

In [7]:
# counting unique companies in companies df
companies['permalink'].str.lower().nunique()

66368

#### 2.3 In the companies data frame, which column can be used as the unique key for each company? Write the name of the column.

In [8]:
# checking the column(s) in the companies df having all unique values
companies.agg(['count', 'size', 'nunique'])

permalink   name  homepage_url  category_list  status  country_code  \
count        66368  66367         61310          63220   66368         59410   
size         66368  66368         66368          66368   66368         66368   
nunique      66368  66102         61191          27296       4           137   

         state_code  region   city  founded_at  
count         57821   58338  58340       51147  
size          66368   66368  66368       66368  
nunique         311    1092   5111        3978

#### 2.4 Are there any companies in the rounds2 file which are not present in companies?

In [9]:
# checking the permalink(s) in rounds2 df that doesn't match those in companies df
print(set(rounds2['company_permalink'].str.lower()) - set(companies['permalink'].str.lower()))

set()


#### 2.5 Merge the two data frames so that all variables (columns) in the companies frame are added to the rounds2 data frame. Name the merged frame master_frame.

In [10]:
# creating function to handle the mixed case strings in 'permalink'/'company_permalink' columns
def case_conv(df, col):                   
    temp = df
    temp[col] = temp[col].str.lower()
    return pd.DataFrame(temp)

# merging the two dataframes
master_frame = pd.merge(case_conv(rounds2, 'company_permalink'), case_conv(companies, 'permalink'), how='left', left_on='company_permalink', right_on='permalink')


In [11]:
# checking the resultant master_frame df
master_frame.head(100)

company_permalink  \
0                        /organization/-fame   
1                     /organization/-qounter   
2                     /organization/-qounter   
3        /organization/-the-one-of-them-inc-   
4                      /organization/0-6-com   
5             /organization/004-technologies   
6           /organization/01games-technology   
7        /organization/0ndine-biomedical-inc   
8        /organization/0ndine-biomedical-inc   
9                       /organization/0xdata   
10                      /organization/0xdata   
11                      /organization/0xdata   
12                      /organization/0xdata   
13                           /organization/1   
14                           /organization/1   
15                           /organization/1   
16                 /organization/1-2-3-listo   
17                     /organization/1-4-all   
18            /organization/1-618-technology   
19               /organization/1-800-dentist   
20               /organization/1-800-doctors   
21  /organization/1-800-publicrelations-inc-   
22                /organization/1-mainstream   
23                     /organization/1-of-99   
24                 /organization/10-20-media   
25                 /organization/10-20-media   
26                 /organization/10-20-media   
27                 /organization/10-20-media   
28             /organization/10-minutes-with   
29             /organization/10-minutes-with   
..                                       ...   
70                       /organization/10six   
71                  /organization/10tracks-2   
72                  /organization/10tracks-2   
73                /organization/10x-genomics   
74            /organization/10x-technologies   
75            /organization/10x-technologies   
76                  /organization/10x10-room   
77                     /organization/10north   
78                   /organization/11-health   
79           /organization/115-network-disks   
80                       /organization/117go   
81               /organization/11i-solutions   
82                     /organization/12-labs   
83            /organization/12-star-survival   
84                  /organization/120-sports   
85                 /organization/121-rentals   
86                     /organization/121cast   
87                     /organization/121cast   
88                     /organization/121cast   
89                    /organization/121nexus   
90                    /organization/121nexus   
91                    /organization/121nexus   
92                    /organization/121nexus   
93                    /organization/121nexus   
94                    /organization/121nexus   
95                   /organization/1234enter   
96                   /organization/1234enter   
97              /organization/123contactform   
98              /organization/123contactform   
99                 /organization/123feng-com   

                            funding_round_permalink   funding_round_type  \
0   /funding-round/9a01d05418af9f794eebff7ace91f638              venture   
1   /funding-round/22dacff496eb7acb2b901dec1dfe5633              venture   
2   /funding-round/b44fbb94153f6cdef13083530bb48030                 seed   
3   /funding-round/650b8f704416801069bb178a1418776b              venture   
4   /funding-round/5727accaeaa57461bd22a9bdd945382d              venture   
5   /funding-round/1278dd4e6a37fa4b7d7e06c21b3c1830              venture   
6   /funding-round/7d53696f2b4f607a2f2a8cbb83d01839          undisclosed   
7   /funding-round/2b9d3ac293d5cdccbecff5c8cb0f327d                 seed   
8   /funding-round/954b9499724b946ad8c396a57a5f3b72              venture   
9   /funding-round/383a9bd2c04f7038bb543ccef5ba3eae                 seed   
10  /funding-round/3bb2ee4a2d89251a10aaa735b1180e44              venture   
11  /funding-round/ae2a174c06517c2394aed45006322a7e              venture   
12  /funding-round/e1cfcbe1bdf4c70277c5f29a3482f24e           

Now before moving ahead to next Checkpoint, it is very important to clean the data in the `master_frame` dataframe

In [12]:
# rename column 'name' to 'company_name'
master_frame.rename(columns={'name' : 'company_name'}, inplace=True)

# remove the redundant column permalink 
master_frame.drop('permalink', axis=1, inplace=True)

In [13]:
# check rows with all NULL values
master_frame.isnull().all(axis=1).sum()

0

In [14]:
# percentage of missing values columnwise
100*master_frame.isnull().sum()/len(master_frame.index)

company_permalink           0.000000
funding_round_permalink     0.000000
funding_round_type          0.000000
funding_round_code         72.909725
funded_at                   0.000000
raised_amount_usd          17.390321
company_name                0.000870
homepage_url                5.336280
category_list               2.966533
status                      0.000000
country_code                7.549435
state_code                  9.522484
region                      8.844792
city                        8.842182
founded_at                 17.852265
dtype: float64

In [15]:
# dropping unwanted columns with missing values
master_frame.drop(['funding_round_code', 'homepage_url', 'founded_at'], axis=1, inplace=True)

In [16]:
# check the no. of rows with null values in raised_amount_usd column
len(master_frame[np.isnan(master_frame['raised_amount_usd'])].index)

19990

In [17]:
# remove all rows with raised_amount_usd as nan
master_frame = master_frame[~ np.isnan(master_frame['raised_amount_usd'])]

# remove row with null in column 'name'
master_frame = master_frame[~ master_frame['company_name'].isnull()]

# percentage of missing values columnwise
100*master_frame.isnull().sum()/len(master_frame.index)

company_permalink          0.000000
funding_round_permalink    0.000000
funding_round_type         0.000000
funded_at                  0.000000
raised_amount_usd          0.000000
company_name               0.000000
category_list              1.099433
status                     0.000000
country_code               6.161671
state_code                 8.007751
region                     7.422229
city                       7.419069
dtype: float64

In [18]:
# let's find out if we can derive country code for the missing rows
# check if all the rows with missing country_code are also missing state_code, region, city
master_frame.loc[master_frame['country_code'].isnull(), 'country_code':'city' ].isnull().all()


country_code    True
state_code      True
region          True
city            True
dtype: bool

In [19]:
# check the rows with null values for country_code column
len(master_frame[master_frame['country_code'].isnull()].index)

5851

In [20]:
# remove rows with null values for country_code
master_frame = master_frame[~ master_frame['country_code'].isnull()]

# percentage of missing values columnwise
100*master_frame.isnull().sum()/len(master_frame.index)

company_permalink          0.000000
funding_round_permalink    0.000000
funding_round_type         0.000000
funded_at                  0.000000
raised_amount_usd          0.000000
company_name               0.000000
category_list              0.649781
status                     0.000000
country_code               0.000000
state_code                 1.967298
region                     1.343329
city                       1.339962
dtype: float64

In [21]:
# remove unwanted state_code, region and city columns, as they have missing values
master_frame.drop(['state_code', 'region', 'city'], axis=1, inplace=True)

# percentage of missing values columnwise
100*master_frame.isnull().sum()/len(master_frame.index)

company_permalink          0.000000
funding_round_permalink    0.000000
funding_round_type         0.000000
funded_at                  0.000000
raised_amount_usd          0.000000
company_name               0.000000
category_list              0.649781
status                     0.000000
country_code               0.000000
dtype: float64

In [22]:
# count the null rows for category_list column
len(master_frame[master_frame['category_list'].isnull() ].index)

579

In [23]:
# remove the rows with null values for category_list
master_frame = master_frame[~ master_frame['category_list'].isnull()]

# percentage of missing values columnwise
100*master_frame.isnull().sum()/len(master_frame.index)

company_permalink          0.0
funding_round_permalink    0.0
funding_round_type         0.0
funded_at                  0.0
raised_amount_usd          0.0
company_name               0.0
category_list              0.0
status                     0.0
country_code               0.0
dtype: float64

Finally, we have no null values in `master_frame` anymore; let's check how much data we're left with and what percentage of the original data we've lost in the process of cleaning.

In [24]:
# check the remaining data
master_frame.shape

(88528, 9)

In [25]:
# percentage of observations left
len(master_frame.index)*100/114949

77.01502405414575

It's good that we still have `~77%` data still left to analyse.

## Checkpoint 2: Funding Type Analysis

### 1. Calculate the average investment amount for each of the four funding types (venture, angel, seed, and private equity) and report the answers in Table 2.1.

In [26]:
# convert the raised_amount_usd to million usd
master_frame['raised_amount_usd'] = (master_frame['raised_amount_usd'])/1000000

# rename the column as raised_amount_m_usd
master_frame.rename(columns={'raised_amount_usd' : 'raised_amount_m_usd'}, inplace=True)

master_frame.head(100)

company_permalink  \
0                         /organization/-fame   
2                      /organization/-qounter   
4                       /organization/0-6-com   
6            /organization/01games-technology   
7         /organization/0ndine-biomedical-inc   
8         /organization/0ndine-biomedical-inc   
9                        /organization/0xdata   
10                       /organization/0xdata   
11                       /organization/0xdata   
12                       /organization/0xdata   
13                            /organization/1   
15                            /organization/1   
16                  /organization/1-2-3-listo   
20                /organization/1-800-doctors   
21   /organization/1-800-publicrelations-inc-   
22                 /organization/1-mainstream   
23                      /organization/1-of-99   
24                  /organization/10-20-media   
25                  /organization/10-20-media   
26                  /organization/10-20-media   
27                  /organization/10-20-media   
28              /organization/10-minutes-with   
29              /organization/10-minutes-with   
30                   /organization/1000-corks   
31                 /organization/1000-markets   
34                 /organization/1000memories   
35                 /organization/1000memories   
36              /organization/1000museums-com   
37              /organization/1000museums-com   
38              /organization/1000museums-com   
..                                        ...   
102                        /organization/1248   
103              /organization/128-technology   
104                       /organization/12bis   
105                        /organization/12cm   
106                    /organization/12return   
108           /organization/1366-technologies   
109           /organization/1366-technologies   
110           /organization/1366-technologies   
111           /organization/1366-technologies   
112           /organization/1366-technologies   
113           /organization/1366-technologies   
114           /organization/1366-technologies   
115           /organization/1366-technologies   
116           /organization/1366-technologies   
118                    /organization/13th-lab   
119                   /organization/140-proof   
120                   /organization/140-proof   
121                     /organization/140fire   
124                      /organization/15five   
125                      /organization/15five   
126                      /organization/15five   
129           /organization/16-mile-solutions   
130                      /organization/169-st   
132                      /organization/16wifi   
134                 /organization/170-systems   
135                      /organization/17hats   
136                      /organization/17hats   
137                      /organization/17u-cn   
138                      /organization/17u-cn   
139                      /organization/17u-cn   

                             funding_round_permalink funding_round_type  \
0    /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
2    /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
4    /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
6    /funding-round/7d53696f2b4f607a2f2a8cbb83d01839        undisclosed   
7    /funding-round/2b9d3ac293d5cdccbecff5c8cb0f327d               seed   
8    /funding-round/954b9499724b946ad8c396a57a5f3b72            venture   
9    /funding-round/383a9bd2c04f7038bb543ccef5ba3eae               seed   
10   /funding-round/3bb2ee4a2d89251a10aaa735b1180e44            venture   
11   /funding-round/ae2a174c06517c2394aed45006322a7e            venture   
12   /funding-round/e1cfcbe1bdf4c70277c5f29a3482f24e            venture   
13   /funding-round/03b975068632eba5bfdb937ec8c07a68               seed   
15   /funding-round/e82464f22241715dd1a6c77241055ed1               seed   
16   /funding-

In [27]:
# finding the average funding per funding_round_type
master_frame.groupby(['funding_round_type']).raised_amount_m_usd.mean()


funding_round_type
angel                      0.971574
convertible_note           1.337187
debt_financing            17.167653
equity_crowdfunding        0.509898
grant                      4.512698
non_equity_assistance      0.480753
post_ipo_debt            169.451790
post_ipo_equity           66.077059
private_equity            73.938486
product_crowdfunding       1.353227
secondary_market          84.438532
seed                       0.747828
undisclosed               15.891661
venture                   11.724223
Name: raised_amount_m_usd, dtype: float64

Clearly, out of the four funding types (angel, seed, venture and private equity) the average amount raised against ```Venture``` which is approximately 11.72 million USD is the one which best fits in Spark Funds' investment budget of 5 to 15 million USD.

## Checkpoint 3: Country Analysis

### 1. Spark Funds wants to see the top nine countries which have received the highest total funding (across ALL sectors for the chosen investment type)

### 2. For the chosen investment type, make a data frame named top9 with the top nine countries (based on the total investment amount each country has received)

In [28]:
# derive a dataframe having all rows with  Venture investment type
venture_frame = master_frame[master_frame['funding_round_type']=='venture']

venture_frame.head()

company_permalink  \
0                   /organization/-fame   
4                 /organization/0-6-com   
8   /organization/0ndine-biomedical-inc   
10                 /organization/0xdata   
11                 /organization/0xdata   

                            funding_round_permalink funding_round_type  \
0   /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
4   /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
8   /funding-round/954b9499724b946ad8c396a57a5f3b72            venture   
10  /funding-round/3bb2ee4a2d89251a10aaa735b1180e44            venture   
11  /funding-round/ae2a174c06517c2394aed45006322a7e            venture   

     funded_at  raised_amount_m_usd            company_name  category_list  \
0   05-01-2015            10.000000                   #fame          Media   
4   19-03-2008             2.000000                 0-6.com    Curated Web   
8   21-12-2009             0.719491  Ondine Biomedical Inc.  Biotechnology   
10  09-11-2015            20.000000                  H2O.ai      Analytics   
11  03-01-2013             1.700000                  H2O.ai      Analytics   

       status country_code  
0   operating          IND  
4   operating          CHN  
8   operating          CAN  
10  operating          USA  
11  operating          USA

In [29]:
# listing top 9 countries with highest total funding for venture investment type
# and assigning them to 'top9' dataframe
top9 = pd.DataFrame(venture_frame.groupby(['country_code']).raised_amount_m_usd.sum()).sort_values('raised_amount_m_usd', ascending=False).head(9)

print(top9)

              raised_amount_m_usd
country_code                     
USA                 420068.029342
CHN                  39338.918773
GBR                  20072.813004
IND                  14261.508718
CAN                   9482.217668
FRA                   7226.851352
ISR                   6854.350477
DEU                   6306.921981
JPN                   3167.647127


### Identify the top three English-speaking countries in the data frame top9.

Clearly, from above ```top9``` dataframe the top 3 countries are ```USA```, ```China```, and ```UK``` but since ```English``` is not an official language in ```China``` therefore ```India``` takes third place after ```USA``` and ```UK```.

## Checkpoint 4: Sector Analysis 1

### 1. Extract the primary sector of each category list from the category_list column

In [30]:
# listing the Primary sector for each row into a new column primary_category
master_frame['primary_category'] = master_frame['category_list'].str.split('|', n=1, expand=False).str[0]

master_frame.head()

company_permalink  \
0                  /organization/-fame   
2               /organization/-qounter   
4                /organization/0-6-com   
6     /organization/01games-technology   
7  /organization/0ndine-biomedical-inc   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
2  /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
6  /funding-round/7d53696f2b4f607a2f2a8cbb83d01839        undisclosed   
7  /funding-round/2b9d3ac293d5cdccbecff5c8cb0f327d               seed   

    funded_at  raised_amount_m_usd            company_name  \
0  05-01-2015             10.00000                   #fame   
2  01-03-2014              0.70000                :Qounter   
4  19-03-2008              2.00000                 0-6.com   
6  01-07-2014              0.04125      01Games Technology   
7  11-09-2009              0.04336  Ondine Biomedical Inc.   

                                       category_list     status country_code  \
0                                              Media  operating          IND   
2  Application Platforms|Real Time|Social Network...  operating          USA   
4                                        Curated Web  operating          CHN   
6                                              Games  operating          HKG   
7                                      Biotechnology  operating          CAN   

        primary_category  
0                  Media  
2  Application Platforms  
4            Curated Web  
6                  Games  
7          Biotechnology

### 2. Use the mapping file 'mapping.csv' to map each primary sector to one of the eight main sectors (Note that ‘Others’ is also considered one of the main sectors)

In [31]:
# load mapping.csv to a new dataframe
sector_mapping = pd.read_csv("datafiles/mapping.csv", encoding = "ISO-8859-1")

sector_mapping.head(10)

category_list  Automotive & Sports  Blanks  \
0                 NaN                    0       1   
1                  3D                    0       0   
2         3D Printing                    0       0   
3       3D Technology                    0       0   
4          Accounting                    0       0   
5    Active Lifestyle                    0       0   
6        Ad Targeting                    0       0   
7  Advanced Materials                    0       0   
8    Adventure Travel                    1       0   
9         Advertising                    0       0   

   Cleantech / Semiconductors  Entertainment  Health  Manufacturing  \
0                           0              0       0              0   
1                           0              0       0              1   
2                           0              0       0              1   
3                           0              0       0              1   
4                           0              0       0              0   
5                           0              0       1              0   
6                           0              0       0              0   
7                           0              0       0              1   
8                           0              0       0              0   
9                           0              0       0              0   

   News, Search and Messaging  Others  Social, Finance, Analytics, Advertising  
0                           0       0                                        0  
1                           0       0                                        0  
2                           0       0                                        0  
3                           0       0                                        0  
4                           0       0                                        1  
5                           0       0                                        0  
6                           0       0                                        1  
7                           0       0                                        0  
8                           0       0                                        0  
9                           0       0                                        1

In [32]:
# checking the data if require cleaning
sector_mapping.loc[sector_mapping.category_list.str.contains('0|1|2|3|4|5|6|7|8|9', na=False), 'category_list'].unique()

array(['3D', '3D Printing', '3D Technology', 'Alter0tive Medicine',
       'A0lytics', 'B2B', 'B2B Express Delivery', 'Big Data A0lytics',
       'Business A0lytics', 'Can0bis', 'Career Ma0gement',
       'Chi0 Internet', 'Cloud Ma0gement', 'Contact Ma0gement',
       'Digital Rights Ma0gement', 'Digital Sig0ge', 'Document Ma0gement',
       'Educatio0l Games', 'Energy Ma0gement', 'Enterprise 2.0',
       'Event Ma0gement', 'Fi0nce', 'Fi0nce Technology',
       'Fi0ncial Exchanges', 'Fi0ncial Services', 'Fleet Ma0gement',
       'Gover0nce', 'Identity Ma0gement', 'Innovation Ma0gement',
       'Intellectual Asset Ma0gement', 'Investment Ma0gement',
       'IT Ma0gement', 'Jour0lism', 'K-12 Education',
       'Knowledge Ma0gement', 'Lead Ma0gement', 'M2M',
       'Medical Professio0ls', 'Mobile A0lytics', '0notechnology',
       '0tural Language Processing', '0tural Resources', '0vigation',
       'P2P Money Transfer', 'Perso0l Branding', 'Perso0l Data',
       'Perso0l Fi0nce', 'Perso0

Now looking at the above situation, this is very clear that there are a number of entries where '0' needs to be replaced with 'na' except for one entry ```Enterprise 2.0```.

In [33]:
# cleaning the errorneous values
sector_mapping.loc[sector_mapping['category_list'] != 'Enterprise 2.0', ['category_list'] ] = sector_mapping['category_list'].str.replace('0', 'na')

# checking the data after cleaning
sector_mapping.loc[sector_mapping.category_list.str.contains('0|1|2|3|4|5|6|7|8|9', na=False), 'category_list'].unique()

array(['3D', '3D Printing', '3D Technology', 'B2B',
       'B2B Express Delivery', 'Enterprise 2.0', 'K-12 Education', 'M2M',
       'P2P Money Transfer'], dtype=object)

Therefore data in ```sector_mapping``` dataframe is clean now. Let us now unpivot the dataframe;

In [34]:
# un-pivoting the dataframe sector_mapping using melt() function
sector_mapping = sector_mapping.melt(id_vars=['category_list'], var_name='main_sector')
sector_mapping.head()

category_list          main_sector  value
0            NaN  Automotive & Sports      0
1             3D  Automotive & Sports      0
2    3D Printing  Automotive & Sports      0
3  3D Technology  Automotive & Sports      0
4     Accounting  Automotive & Sports      0

Now we can safely remove the rows with value '0'

In [35]:
# removing unwanted rows with value=0
sector_mapping = sector_mapping[~(sector_mapping['value']==0)]

# removing the value column as not required anymore
sector_mapping.drop('value', axis=1, inplace=True)

sector_mapping.head()

category_list          main_sector
8   Adventure Travel  Automotive & Sports
14         Aerospace  Automotive & Sports
45              Auto  Automotive & Sports
46   Automated Kiosk  Automotive & Sports
47        Automotive  Automotive & Sports

Before merging two dataframes, one last thing is to ensure the perfect merge by converting all values of key columns to same case,

In [36]:
# for sector_mapping dataframe
sector_mapping['category_list'] = sector_mapping['category_list'].str.lower()

# for master_frame dataframe
master_frame['primary_category'] = master_frame['primary_category'].str.lower()

Thus, we are good to map the two dataframes and achieve the column ```main_sector``` into the ```master_frame```

In [37]:
master_frame = master_frame.merge(sector_mapping, how='left', left_on='primary_category', right_on='category_list')

master_frame.loc[:,:]

company_permalink  \
0                                    /organization/-fame   
1                                 /organization/-qounter   
2                                  /organization/0-6-com   
3                       /organization/01games-technology   
4                    /organization/0ndine-biomedical-inc   
5                    /organization/0ndine-biomedical-inc   
6                                   /organization/0xdata   
7                                   /organization/0xdata   
8                                   /organization/0xdata   
9                                   /organization/0xdata   
10                                       /organization/1   
11                                       /organization/1   
12                             /organization/1-2-3-listo   
13                           /organization/1-800-doctors   
14              /organization/1-800-publicrelations-inc-   
15                            /organization/1-mainstream   
16                                 /organization/1-of-99   
17                             /organization/10-20-media   
18                             /organization/10-20-media   
19                             /organization/10-20-media   
20                             /organization/10-20-media   
21                         /organization/10-minutes-with   
22                         /organization/10-minutes-with   
23                              /organization/1000-corks   
24                            /organization/1000-markets   
25                            /organization/1000memories   
26                            /organization/1000memories   
27                         /organization/1000museums-com   
28                         /organization/1000museums-com   
29                         /organization/1000museums-com   
...                                                  ...   
88498                              /organization/zynstra   
88499                              /organization/zynstra   
88500                              /organization/zynstra   
88501                              /organization/zynstra   
88502                           /organization/zyomyx-inc   
88503                           /organization/zyomyx-inc   
88504                           /organization/zyomyx-inc   
88505                           /organization/zyomyx-inc   
88506                                 /organization/zype   
88507                                 /organization/zype   
88508                               /organization/zypsee   
88509                     /organization/zyraz-technology   
88510                     /organization/zyraz-technology   
88511                     /organization/zyraz-technology   
88512                     /organization/zyraz-technology   
88513                                /organization/zyrra   
88514                                /organization/zyrra   
88515                                /organization/zyrra   
88516                                /organization/zyrra   
88517                               /organization/zystor   
88518                           /organization/zytoprotec   
88519                                /organization/zzish   
88520                                /organization/zzish   
88521                                /organization/zzish   
88522  /organization/zznode-science-and-technology-co...   
88523                          /organization/zzzzapp-com   
88524                          /organization/zzzzapp-com   
88525                          /organization/zzzzapp-com   
88526                          /organization/zzzzapp-com   
88527                               /organization/asys-2   

                               funding_round_permalink funding_round_type  \
0      /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1      /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
2      /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
3      /fu

Hence, we got the required merged frame with all the details together, let us now clean the dataframe a bit by removing the redundant ```category_list_y``` column and rename column ```category_list_x``` to remove '_x'

In [38]:
# remove redundant column 
master_frame.drop('category_list_y', axis=1, inplace=True)

# rename column category_list_x 
master_frame.rename(columns={'category_list_x' : 'category_list'}, inplace=True)

The final ```master_frame``` with ```primary_category``` for each row mapped to its ```main_sector``` is ready as below,

In [39]:
master_frame.loc[:,:]

company_permalink  \
0                                    /organization/-fame   
1                                 /organization/-qounter   
2                                  /organization/0-6-com   
3                       /organization/01games-technology   
4                    /organization/0ndine-biomedical-inc   
5                    /organization/0ndine-biomedical-inc   
6                                   /organization/0xdata   
7                                   /organization/0xdata   
8                                   /organization/0xdata   
9                                   /organization/0xdata   
10                                       /organization/1   
11                                       /organization/1   
12                             /organization/1-2-3-listo   
13                           /organization/1-800-doctors   
14              /organization/1-800-publicrelations-inc-   
15                            /organization/1-mainstream   
16                                 /organization/1-of-99   
17                             /organization/10-20-media   
18                             /organization/10-20-media   
19                             /organization/10-20-media   
20                             /organization/10-20-media   
21                         /organization/10-minutes-with   
22                         /organization/10-minutes-with   
23                              /organization/1000-corks   
24                            /organization/1000-markets   
25                            /organization/1000memories   
26                            /organization/1000memories   
27                         /organization/1000museums-com   
28                         /organization/1000museums-com   
29                         /organization/1000museums-com   
...                                                  ...   
88498                              /organization/zynstra   
88499                              /organization/zynstra   
88500                              /organization/zynstra   
88501                              /organization/zynstra   
88502                           /organization/zyomyx-inc   
88503                           /organization/zyomyx-inc   
88504                           /organization/zyomyx-inc   
88505                           /organization/zyomyx-inc   
88506                                 /organization/zype   
88507                                 /organization/zype   
88508                               /organization/zypsee   
88509                     /organization/zyraz-technology   
88510                     /organization/zyraz-technology   
88511                     /organization/zyraz-technology   
88512                     /organization/zyraz-technology   
88513                                /organization/zyrra   
88514                                /organization/zyrra   
88515                                /organization/zyrra   
88516                                /organization/zyrra   
88517                               /organization/zystor   
88518                           /organization/zytoprotec   
88519                                /organization/zzish   
88520                                /organization/zzish   
88521                                /organization/zzish   
88522  /organization/zznode-science-and-technology-co...   
88523                          /organization/zzzzapp-com   
88524                          /organization/zzzzapp-com   
88525                          /organization/zzzzapp-com   
88526                          /organization/zzzzapp-com   
88527                               /organization/asys-2   

                               funding_round_permalink funding_round_type  \
0      /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1      /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
2      /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
3      /fu

Now, it is good to check if there are any NULL values in our dataframe,

In [40]:
# check for any null values in the master_frame
master_frame.isnull().sum()

company_permalink           0
funding_round_permalink     0
funding_round_type          0
funded_at                   0
raised_amount_m_usd         0
company_name                0
category_list               0
status                      0
country_code                0
primary_category            0
main_sector                63
dtype: int64

So we have 63 rows with missing main_sector details, let us see which primary categories are those,

In [41]:
# print the primary_categories which are missing main_sector values
print(master_frame.loc[master_frame['main_sector'].isnull(), :].primary_category.unique())
print('\n')
# print their number
print(len(master_frame.loc[master_frame['main_sector'].isnull(), :].primary_category.unique()))

['cause marketing' 'real estate investors' 'english-speaking' 'toys'
 'generation y-z' 'enterprise hardware' 'social media advertising'
 'natural gas uses' 'internet technology' 'nightlife' 'adaptive equipment'
 'internet tv' 'skill gaming' 'racing' 'specialty retail' 'swimming'
 'registrars' 'golf equipment' 'biotechnology and semiconductor'
 'vacation rentals' 'google glass' 'rapidly expanding' 'group email'
 'kinect' 'spas' 'product search' 'sex industry' 'psychology' 'greentech'
 'retirement' 'subscription businesses' 'lingerie'
 'mobile emergency&health' 'sponsorship' 'deep information technology']


35


There is no use of keeping these 63 rows as we're missing the analysis parameter(sector) for these 63 rows.

In [42]:
# remove the rows with null values
master_frame = master_frame[~ master_frame['main_sector'].isnull()]

# print the primary_categories which are missing main_sector values
print(master_frame.loc[master_frame['main_sector'].isnull(), :].primary_category.unique())

[]


Now, we are here with the clean and final dataframe

In [43]:
master_frame.loc[:,:]

company_permalink  \
0                                    /organization/-fame   
1                                 /organization/-qounter   
2                                  /organization/0-6-com   
3                       /organization/01games-technology   
4                    /organization/0ndine-biomedical-inc   
5                    /organization/0ndine-biomedical-inc   
6                                   /organization/0xdata   
7                                   /organization/0xdata   
8                                   /organization/0xdata   
9                                   /organization/0xdata   
10                                       /organization/1   
11                                       /organization/1   
12                             /organization/1-2-3-listo   
13                           /organization/1-800-doctors   
14              /organization/1-800-publicrelations-inc-   
15                            /organization/1-mainstream   
16                                 /organization/1-of-99   
17                             /organization/10-20-media   
18                             /organization/10-20-media   
19                             /organization/10-20-media   
20                             /organization/10-20-media   
21                         /organization/10-minutes-with   
22                         /organization/10-minutes-with   
23                              /organization/1000-corks   
24                            /organization/1000-markets   
25                            /organization/1000memories   
26                            /organization/1000memories   
27                         /organization/1000museums-com   
28                         /organization/1000museums-com   
29                         /organization/1000museums-com   
...                                                  ...   
88498                              /organization/zynstra   
88499                              /organization/zynstra   
88500                              /organization/zynstra   
88501                              /organization/zynstra   
88502                           /organization/zyomyx-inc   
88503                           /organization/zyomyx-inc   
88504                           /organization/zyomyx-inc   
88505                           /organization/zyomyx-inc   
88506                                 /organization/zype   
88507                                 /organization/zype   
88508                               /organization/zypsee   
88509                     /organization/zyraz-technology   
88510                     /organization/zyraz-technology   
88511                     /organization/zyraz-technology   
88512                     /organization/zyraz-technology   
88513                                /organization/zyrra   
88514                                /organization/zyrra   
88515                                /organization/zyrra   
88516                                /organization/zyrra   
88517                               /organization/zystor   
88518                           /organization/zytoprotec   
88519                                /organization/zzish   
88520                                /organization/zzish   
88521                                /organization/zzish   
88522  /organization/zznode-science-and-technology-co...   
88523                          /organization/zzzzapp-com   
88524                          /organization/zzzzapp-com   
88525                          /organization/zzzzapp-com   
88526                          /organization/zzzzapp-com   
88527                               /organization/asys-2   

                               funding_round_permalink funding_round_type  \
0      /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1      /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
2      /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
3      /fu

## Checkpoint 5: Sector Analysis 2

### 1. Create three separate data frames D1, D2 and D3 for each of the three countries containing the observations of funding type FT falling within the 5-15 million USD range. The three data frames should contain:

*  All the columns of the master_frame along with the primary sector and the main sector
*  The total number (or count) of investments for each main sector in a separate column
*  The total amount invested in each main sector in a separate column

In [44]:
# creating dataframes D1, D2 and D3 for USA, GBR and IND with all columns from master_frame 
# having Venture funding type and falling within 5-15 million USD funding range.
D1 = master_frame.loc[((master_frame.country_code=='USA') & (master_frame.funding_round_type=='venture') & master_frame.raised_amount_m_usd.between(5,15)), :]

D2 = master_frame.loc[((master_frame.country_code=='GBR') & (master_frame.funding_round_type=='venture') & master_frame.raised_amount_m_usd.between(5,15)), :]

D3 = master_frame.loc[((master_frame.country_code=='IND') & (master_frame.funding_round_type=='venture') & master_frame.raised_amount_m_usd.between(5,15)), :]


In [45]:
# check D1 for USA
D1.head()

company_permalink  \
9              /organization/0xdata   
15       /organization/1-mainstream   
71     /organization/128-technology   
77  /organization/1366-technologies   
78  /organization/1366-technologies   

                            funding_round_permalink funding_round_type  \
9   /funding-round/e1cfcbe1bdf4c70277c5f29a3482f24e            venture   
15  /funding-round/b952cbaf401f310927430c97b68162ea            venture   
71  /funding-round/fb6216a30cb566ede89e0bee0623a634            venture   
77  /funding-round/424129ce1235cfab2655ee81305f7c2b            venture   
78  /funding-round/6d3f3797371956ece035b8478c1441b2            venture   

     funded_at  raised_amount_m_usd       company_name  \
9   19-07-2014             8.900000             H2O.ai   
15  17-03-2015             5.000000       1 Mainstream   
71  16-12-2014            11.999347     128 Technology   
77  15-10-2013            15.000000  1366 Technologies   
78  09-04-2015             5.000000  1366 Technologies   

                       category_list     status country_code  \
9                          Analytics  operating          USA   
15  Apps|Cable|Distribution|Software   acquired          USA   
71      Service Providers|Technology  operating          USA   
77                     Manufacturing  operating          USA   
78                     Manufacturing  operating          USA   

     primary_category                              main_sector  
9           analytics  Social, Finance, Analytics, Advertising  
15               apps               News, Search and Messaging  
71  service providers                                   Others  
77      manufacturing                            Manufacturing  
78      manufacturing                            Manufacturing

In [46]:
# check D2 for Great Britain/UK
D2.head()

company_permalink                          funding_round_permalink  \
289  /organization/365scores  /funding-round/48212f931f542fdef78810bc87aef086   
623   /organization/7digital  /funding-round/b5ad7ed7baddd3974bd51403f17dd88f   
624   /organization/7digital  /funding-round/eafacfcceb1fbc4fd605f641b603313e   
677      /organization/90min  /funding-round/21a2cbf6f2fb2a1c2a61e04bf930dfe6   
678      /organization/90min  /funding-round/bd626ed022f5c66574b1afe234f3c90d   

    funding_round_type   funded_at  raised_amount_m_usd company_name  \
289            venture  29-09-2014             5.500000    365Scores   
623            venture  01-01-2008             8.468328     7digital   
624            venture  19-10-2012            10.000000     7digital   
677            venture  06-10-2015            15.000000        90min   
678            venture  07-05-2013             5.800000        90min   

                                         category_list     status  \
289                  Android|Apps|iPhone|Mobile|Sports  operating   
623  Content Creators|Content Delivery|Licensing|Mu...   acquired   
624  Content Creators|Content Delivery|Licensing|Mu...   acquired   
677                Media|News|Publishing|Soccer|Sports  operating   
678                Media|News|Publishing|Soccer|Sports  operating   

    country_code  primary_category                              main_sector  
289          GBR           android  Social, Finance, Analytics, Advertising  
623          GBR  content creators                            Entertainment  
624          GBR  content creators                            Entertainment  
677          GBR             media                            Entertainment  
678          GBR             media                            Entertainment

In [47]:
# check D3 for India
D3.head()

company_permalink  \
0                /organization/-fame   
162   /organization/21diamonds-india   
753     /organization/a-little-world   
1920    /organization/adlabs-imagica   
2665             /organization/agile   

                              funding_round_permalink funding_round_type  \
0     /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
162   /funding-round/6de7ffef8091ba9f33821f4b861f434a            venture   
753   /funding-round/18d98f82ed392b1609975b81f3e8b3fb            venture   
1920  /funding-round/508d3c83daaae9fda3ba6f9682c78f6c            venture   
2665  /funding-round/cd3dd1c98ce9d0f632d8752163941674            venture   

       funded_at  raised_amount_m_usd    company_name  \
0     05-01-2015            10.000000           #fame   
162   15-11-2012             6.369507      21Diamonds   
753   09-09-2008             6.410000  A LITTLE WORLD   
1920  28-10-2014             8.180000  Adlabs Imagica   
2665  01-05-2011             5.740000           Agile   

                                     category_list     status country_code  \
0                                            Media  operating          IND   
162                                     E-Commerce  operating          IND   
753                                        Finance  operating          IND   
1920                         Entertainment|Tourism  operating          IND   
2665  Finance|Finance Technology|FinTech|Insurance  operating          IND   

     primary_category                              main_sector  
0               media                            Entertainment  
162        e-commerce                                   Others  
753           finance  Social, Finance, Analytics, Advertising  
1920    entertainment                            Entertainment  
2665          finance  Social, Finance, Analytics, Advertising

Next step is to add column say ```no_of_inv``` to all three dataframes.

In [49]:
# adding no_of_inv column to D1
D1['no_of_inv'] = D1.groupby('main_sector').main_sector.transform('count')

# adding no_of_inv column to D2
D2['no_of_inv'] = D2.groupby('main_sector').main_sector.transform('count')

# adding no_of_inv column to D3
D3['no_of_inv'] = D3.groupby('main_sector').main_sector.transform('count')

And add column say ```amt_of_inv``` to all three dataframes.

In [52]:
# adding amt_of_inv to D1
D1.loc[:,'amt_of_inv'] = D1.groupby('main_sector').raised_amount_m_usd.transform('sum')

# adding amt_of_inv to D2
D2.loc[:,'amt_of_inv'] = D2.groupby('main_sector').raised_amount_m_usd.transform('sum')

# adding amt_of_inv to D3
D3.loc[:,'amt_of_inv'] = D3.groupby('main_sector').raised_amount_m_usd.transform('sum')

Therefore, the final three dataframes D1, D2 and D3 are as below

In [53]:
# check D1 for USA
D1.head()

company_permalink  \
9              /organization/0xdata   
15       /organization/1-mainstream   
71     /organization/128-technology   
77  /organization/1366-technologies   
78  /organization/1366-technologies   

                            funding_round_permalink funding_round_type  \
9   /funding-round/e1cfcbe1bdf4c70277c5f29a3482f24e            venture   
15  /funding-round/b952cbaf401f310927430c97b68162ea            venture   
71  /funding-round/fb6216a30cb566ede89e0bee0623a634            venture   
77  /funding-round/424129ce1235cfab2655ee81305f7c2b            venture   
78  /funding-round/6d3f3797371956ece035b8478c1441b2            venture   

     funded_at  raised_amount_m_usd       company_name  \
9   19-07-2014             8.900000             H2O.ai   
15  17-03-2015             5.000000       1 Mainstream   
71  16-12-2014            11.999347     128 Technology   
77  15-10-2013            15.000000  1366 Technologies   
78  09-04-2015             5.000000  1366 Technologies   

                       category_list     status country_code  \
9                          Analytics  operating          USA   
15  Apps|Cable|Distribution|Software   acquired          USA   
71      Service Providers|Technology  operating          USA   
77                     Manufacturing  operating          USA   
78                     Manufacturing  operating          USA   

     primary_category                              main_sector  no_of_inv  \
9           analytics  Social, Finance, Analytics, Advertising       2714   
15               apps               News, Search and Messaging       1583   
71  service providers                                   Others       2950   
77      manufacturing                            Manufacturing        799   
78      manufacturing                            Manufacturing        799   

      amt_of_inv  
9   23807.376964  
15  13971.567428  
71  26321.007002  
77   7258.553378  
78   7258.553378

In [54]:
# check D2 for Great Britain/UK
D2.head()

company_permalink                          funding_round_permalink  \
289  /organization/365scores  /funding-round/48212f931f542fdef78810bc87aef086   
623   /organization/7digital  /funding-round/b5ad7ed7baddd3974bd51403f17dd88f   
624   /organization/7digital  /funding-round/eafacfcceb1fbc4fd605f641b603313e   
677      /organization/90min  /funding-round/21a2cbf6f2fb2a1c2a61e04bf930dfe6   
678      /organization/90min  /funding-round/bd626ed022f5c66574b1afe234f3c90d   

    funding_round_type   funded_at  raised_amount_m_usd company_name  \
289            venture  29-09-2014             5.500000    365Scores   
623            venture  01-01-2008             8.468328     7digital   
624            venture  19-10-2012            10.000000     7digital   
677            venture  06-10-2015            15.000000        90min   
678            venture  07-05-2013             5.800000        90min   

                                         category_list     status  \
289                  Android|Apps|iPhone|Mobile|Sports  operating   
623  Content Creators|Content Delivery|Licensing|Mu...   acquired   
624  Content Creators|Content Delivery|Licensing|Mu...   acquired   
677                Media|News|Publishing|Soccer|Sports  operating   
678                Media|News|Publishing|Soccer|Sports  operating   

    country_code  primary_category                              main_sector  \
289          GBR           android  Social, Finance, Analytics, Advertising   
623          GBR  content creators                            Entertainment   
624          GBR  content creators                            Entertainment   
677          GBR             media                            Entertainment   
678          GBR             media                            Entertainment   

     no_of_inv   amt_of_inv  
289        133  1089.404014  
623         56   482.784687  
624         56   482.784687  
677         56   482.784687  
678         56   482.784687

In [55]:
# check D3 for India
D3.head()

company_permalink  \
0                /organization/-fame   
162   /organization/21diamonds-india   
753     /organization/a-little-world   
1920    /organization/adlabs-imagica   
2665             /organization/agile   

                              funding_round_permalink funding_round_type  \
0     /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
162   /funding-round/6de7ffef8091ba9f33821f4b861f434a            venture   
753   /funding-round/18d98f82ed392b1609975b81f3e8b3fb            venture   
1920  /funding-round/508d3c83daaae9fda3ba6f9682c78f6c            venture   
2665  /funding-round/cd3dd1c98ce9d0f632d8752163941674            venture   

       funded_at  raised_amount_m_usd    company_name  \
0     05-01-2015            10.000000           #fame   
162   15-11-2012             6.369507      21Diamonds   
753   09-09-2008             6.410000  A LITTLE WORLD   
1920  28-10-2014             8.180000  Adlabs Imagica   
2665  01-05-2011             5.740000           Agile   

                                     category_list     status country_code  \
0                                            Media  operating          IND   
162                                     E-Commerce  operating          IND   
753                                        Finance  operating          IND   
1920                         Entertainment|Tourism  operating          IND   
2665  Finance|Finance Technology|FinTech|Insurance  operating          IND   

     primary_category                              main_sector  no_of_inv  \
0               media                            Entertainment         33   
162        e-commerce                                   Others        110   
753           finance  Social, Finance, Analytics, Advertising         60   
1920    entertainment                            Entertainment         33   
2665          finance  Social, Finance, Analytics, Advertising         60   

       amt_of_inv  
0      280.830000  
162   1013.409507  
753    550.549550  
1920   280.830000  
2665   550.549550

### 2. Table 5.1: Based on the analysis of the sectors, which main sectors and countries would you recommend Spark Funds to invest in? Present your conclusions in the presentation. The conclusions are subjective (i.e. there may be no ‘one right answer’), but it should be based on the basic strategy — invest in sectors where most investments are occurring. 

#### 1.  Total number of Investments (count)

In [56]:
# total no of investments in USA
print("Total investments in USA : " + str(D1.no_of_inv.unique().sum()))

# total no of investments in UK
print("Total investments in GBR : " + str(D2.no_of_inv.unique().sum()))

# total no of investments in India
print("Total investments in India : " + str(D3.no_of_inv.unique().sum()))

Total investments in USA : 12063
Total investments in GBR : 621
Total investments in India : 328



#### 2.  Total amount of investment (USD)

In [57]:
# total amount(USD) invested in USA
print("Total amount invested in USA : " + str(D1.amt_of_inv.unique().sum()*1000000))

# total amount(USD) invested in UK
print("Total amount invested in GBR : " + str(D2.amt_of_inv.unique().sum()*1000000))

# total amount(USD) invested in India
print("Total amount invested in India : " + str(D3.amt_of_inv.unique().sum()*1000000))

Total amount invested in USA : 107757097293.99998
Total amount invested in GBR : 5379078691.000001
Total amount invested in India : 2949543601.9999995


#### 3.  Top Sector name (no. of investment-wise)
#### 4.  Second Sector name (no. of investment-wise)
#### 5.  Third Sector name (no. of investment-wise)

In [58]:
# Top three sectors in terms of no. of investments in USA
print("\nTop three sectors (no. of investment-wise) in USA : " )
print(pd.DataFrame(D1.groupby('main_sector').no_of_inv.count()).sort_values('no_of_inv', ascending=False).head(3))

# Top three sectors in terms of no. of investments in UK
print("\nTop three sectors (no. of investment-wise) in UK : " )
print(pd.DataFrame(D2.groupby('main_sector').no_of_inv.count()).sort_values('no_of_inv', ascending=False).head(3))

# Top three sectors in terms of no. of investments in India
print("\nTop three sectors (no. of investment-wise) in India : " )
print(pd.DataFrame(D3.groupby('main_sector').no_of_inv.count()).sort_values('no_of_inv', ascending=False).head(3))


Top three sectors (no. of investment-wise) in USA : 
                                         no_of_inv
main_sector                                       
Others                                        2950
Social, Finance, Analytics, Advertising       2714
Cleantech / Semiconductors                    2350

Top three sectors (no. of investment-wise) in UK : 
                                         no_of_inv
main_sector                                       
Others                                         147
Social, Finance, Analytics, Advertising        133
Cleantech / Semiconductors                     130

Top three sectors (no. of investment-wise) in India : 
                                         no_of_inv
main_sector                                       
Others                                         110
Social, Finance, Analytics, Advertising         60
News, Search and Messaging                      52


Next three questions are already answered in the above section.

#### 6.  Number of investments in top sector (3)
#### 7.  Number of investments in second sector (4)
#### 8.  Number of investments in third sector (5)


#### 9.  For point 3 (top sector count-wise), which company received the highest investment?

In [59]:
# Top 5 'USA' companies of 'Others' sector to get investment 
pd.DataFrame(D1.loc[(D1.main_sector=='Others'), :].groupby('company_name')['raised_amount_m_usd'].sum()).sort_values(by='raised_amount_m_usd', ascending=False).head()

raised_amount_m_usd
company_name                            
Virtustream                    64.300000
Capella Photonics              54.968051
AirTight Networks              54.201907
deCarta                        52.100000
Black Duck Software            51.000000

In [60]:
# Top 5 'UK' companies of 'Others' sector to get investment 
pd.DataFrame(D2.loc[(D2.main_sector=='Others'), :].groupby('company_name')['raised_amount_m_usd'].sum()).sort_values(by='raised_amount_m_usd', ascending=False).head()

raised_amount_m_usd
company_name                                
Electric Cloud                     37.000000
SenSage                            36.250000
Enigmatec                          32.500000
SilverRail Technologies            29.000000
OpenCloud                          27.972766

In [61]:
# Top 5 'Indian' companies of 'Others' sector to get investment 
pd.DataFrame(D3.loc[(D3.main_sector=='Others'), :].groupby('company_name')['raised_amount_m_usd'].sum()).sort_values(by='raised_amount_m_usd', ascending=False).head()

raised_amount_m_usd
company_name                          
FirstCry.com                      39.0
Myntra                            38.0
CommonFloor                       32.9
Pepperfry.com                     28.0
ItzCash Card Ltd.                 25.0

Therefore the top companies to get investments in the top sector for each country `USA`, `GBR` and `IND` are `Virtustream`, `Electric Cloud` and `FirstCry.com` respectively.

#### 10.  For point 4 (second best sector count-wise), which company received the highest investment?

In [62]:
# Top 5 'USA' companies of 'Social, Finance, Analytics, Advertising' sector to get investment 
pd.DataFrame(D1.loc[(D1.main_sector=='Social, Finance, Analytics, Advertising'), :].groupby('company_name')['raised_amount_m_usd'].sum()).sort_values(by='raised_amount_m_usd', ascending=False).head()

raised_amount_m_usd
company_name                                        
SST Inc. (Formerly ShotSpotter)            67.933006
Demandbase                                 63.000000
Intacct                                    61.800000
NetBase Solutions                          60.600000
Lotame                                     59.700000

In [63]:
# Top 5 'UK' companies of 'Social, Finance, Analytics, Advertising' sector to get investment 
pd.DataFrame(D2.loc[(D2.main_sector=='Social, Finance, Analytics, Advertising'), :].groupby('company_name')['raised_amount_m_usd'].sum()).sort_values(by='raised_amount_m_usd', ascending=False).head()

raised_amount_m_usd
company_name                              
Celltick Technologies            37.500000
myThings                         34.000000
Zopa                             32.900000
VisualDNA                        28.550000
MarketInvoice                    25.553007

In [64]:
# Top 5 'Indian' companies of 'Social, Finance, Analytics, Advertising' sector to get investment 
pd.DataFrame(D3.loc[(D3.main_sector=='Social, Finance, Analytics, Advertising'), :].groupby('company_name')['raised_amount_m_usd'].sum()).sort_values(by='raised_amount_m_usd', ascending=False).head()

raised_amount_m_usd
company_name                                   
Manthan Systems                        50.70000
Komli Media                            28.00000
ShopClues.com                          25.00000
inTarvo                                21.90000
Grameen Financial Services             21.55605

And the top companies to get investments in the second sector for each country `USA`, `GBR` and `IND` are `SST Inc. (Formerly ShotSpotter)`, `Celltick Technologies` and `Manthan Systems` respectively.

In [65]:
master_frame.to_excel('master_frame.xlsx')